packages/flutter/lib/src/rendering/table.dart

```diff
@@ -586,124 +558,47 @@ class RenderTable extends RenderBox {
       child.parentData = TableCellParentData();
   }
 
-  /// Replaces the children of this table with the given cells.
+  /// Insert child into this render object's child list after the given child.
   ///
-  /// The cells are divided into the specified number of columns before
-  /// replacing the existing children.
-  ///
-  /// If the new cells contain any existing children of the table, those
-  /// children are simply moved to their new location in the table rather than
-  /// removed from the table and re-added.
-  void setFlatChildren(int columns, List<RenderBox?> cells) {
-    if (cells == _children && columns == _columns)
-      return;
-    assert(columns >= 0);
-    // consider the case of a newly empty table
-    if (columns == 0 || cells.isEmpty) {
-      assert(cells == null || cells.isEmpty);
-      _columns = columns;
-      if (_children.isEmpty) {
-        assert(_rows == 0);
-        return;
-      }
-      for (final RenderBox? oldChild in _children) {
-        if (oldChild != null)
-          dropChild(oldChild);
-      }
-      _rows = 0;
-      _children.clear();
-      markNeedsLayout();
-      return;
+  /// If `after` is null, then this inserts the child at the start of the list.
+  /// Children are stored in row-major order, you should call [setDimensions]
+  /// after updating children is finish to set the table dimensions.
+  void insertChild(RenderBox child, { RenderBox? after }) {
```

Comment:
it would be good to say in the documentation here that using the `after` argument is extremely expensive (potentially O(N^2) depending on how List is implemented, and potentially O(N^3) if used in a loop, which seems like a common usage).

Unless there's a really strong reason for having `after` (I haven't checked the rest of the PR yet) it might be better to just not have that argument available.